server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -1002,19 +1002,6 @@ static void validateSnapshotRestorable(RestoreSnapshotRequest request, Repositor
         Setting.Property.NodeScope
     );
 
-    private static boolean skipVersionChecks(RepositoryMetadata repositoryMetadata) {
-        if (Build.CURRENT.isSnapshot()) {
```

Comment:
I've removed this condition as part of this PR. As we're going to release the feature in 8.2.0, no need to keep this condition.